lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -111,6 +111,10 @@ function findDifference(path, actual, expected) {
     return null;
   }
 
+  if (expected instanceof RegExp && expected.test(actual)) {
```

Comment:
Although there is also the minuscule chance that `actual.toString()` could yield a string that somehow passes `expected.test()`, so you may also want to add a `typeof actual === 'string'` above and below (or rearrange somehow) if that's not a feature we want to support :)